src/QMCHamiltonians/SOECPComponent.cpp

```diff
@@ -140,105 +154,84 @@ SOECPComponent::ComplexType SOECPComponent::getAngularIntegral(RealType sold,
-      psiratio[j] = Psi.calcRatio(W, iel);
+      W.makeMoveWithSpin(iel, deltaV[iq], deltaS[iq]);
+      psiratio[iq] = Psi.calcRatio(W, iel);
       W.rejectMove(iel);
       Psi.resetPhaseDiff();
     }
   }
 
-  //Need to add appropriate weight to psiratio
-  std::transform(psiratio.begin(), psiratio.end(), sgridweight_m.begin(), psiratio.begin(),
-                 [](auto& psi, auto& weight) {
-                   RealType fourpi = 2.0 * TWOPI;
-                   return psi * weight * fourpi;
-                 });
-
-  ComplexType angint(0.0);
-  for (int j = 0; j < nknot; j++)
+  ComplexType pairpot;
+  for (int iq = 0; iq < total_knots; iq++)
   {
-    ComplexType lsum(0.0);
+    RealType snew = sold + deltaS[iq];
+    ComplexType lsum;
     for (int il = 0; il < nchannel; il++)
     {
       int l = il + 1; //nchannels starts at l=1, so 0th element is p not s
-      ComplexType msums(0.0);
+      ComplexType msums;
       for (int m1 = -l; m1 <= l; m1++)
       {
+        ComplexType Y = sphericalHarmonic(l, m1, dr);
         for (int m2 = -l; m2 <= l; m2++)
         {
-          ComplexType ldots(0.0);
-          for (int d = 0; d < 3; d++)
-            ldots += lmMatrixElements(l, m1, m2, d) * sMatrixElements(sold, snew, d);
-          ComplexType Y  = sphericalHarmonic(l, m1, dr);
-          ComplexType cY = std::conj(sphericalHarmonic(l, m2, rrotsgrid_m[j]));
+          ComplexType ldots;
+          for (int id = 0; id < 3; id++)
+            ldots += lmMatrixElements(l, m1, m2, id) * sMatrixElements(sold, snew, id);
+          ComplexType cY = std::conj(sphericalHarmonic(l, m2, rrotsgrid_m[iq % nknot]));
           msums += Y * cY * ldots;
         }
       }
       lsum += vrad[il] * msums;
     }
-    angint += psiratio[j] * lsum;
+    pairpot += psiratio[iq] * lsum * spin_quad_weights[iq];
   }
-  return angint;
+  return std::real(pairpot);
 }
 
-SOECPComponent::RealType SOECPComponent::evaluateOne(ParticleSet& W,
-                                                     int iat,
-                                                     TrialWaveFunction& Psi,
-                                                     int iel,
-                                                     RealType r,
-                                                     const PosType& dr)
+void SOECPComponent::buildTotalQuadrature(const RealType r, const PosType& dr, const RealType sold)
 {
-  if (sknot < 2)
-    APP_ABORT("Spin knots must be greater than 2\n");
-
-  if (sknot % 2 != 0)
-    APP_ABORT("Spin knots uses Simpson's rule. Must have even number of knots");
-
-  for (int ip = 0; ip < nchannel; ip++)
-  {
-    vrad[ip] = sopp_m[ip]->splint(r);
-  }
-
+  int count = 0;
   RealType smin(0.0);
   RealType smax(TWOPI);
   RealType dS = (smax - smin) / sknot; //step size for spin
 
-  RealType sold = W.spins[iel];
-  ComplexType sint(0.0);
+  auto addSpatialQuadrature = [&](const int is, const RealType r, const PosType& dr, const RealType ds,
```

Comment:
Can you add a function comment here explaining what it's doing?  High level, plus what member variables get modified and how.  Also, there's an implied layout of quadrature grids w.r.t. real space and spin quadrature components for example.

Reply:
done. added a description of the lambda and a more detailed comment in the header describing buildTotalQuadrature

Reply:
also, changed all private data members of SOECPComponent to have the trailing _. Note I had to touch both the ECPotentialBuilder and test_ecp since both of those are friend classes that touched some of these members